### PR TITLE
Fix #16568 - Multiple Error Notifications and report requests

### LINKED ISSUE #16568: Error reporting feature submits reports without consent
### Describe the bug

The error reporting feature appears to be written to only handle a single JS error at a time.
However, due to race conditions in the code, it appears to unintentionally submit reports without warning.

### To Reproduce

1. Checkout revision a6422ced622b9e39024f09d4148d65f429b9aa4a
2. Select any table that has some data
3. Hover over a row header a few times to trigger bug #16558

### Expected behavior

* A single error notification is visible at the bottom of the page
* A single  request is sent to the backend server to retrieve user preferences for error reporting
* Clicking on 'Show report details' does not submit the report

### Actual behaviour

* Multiple notification are visible at the bottom of the page
* For every single exception being thrown, a request is sent to the backend server to retrieve user preferences
* Clicking on 'Show report details' submits multiple reports to the backend server

### Screenshots

Here you can see the client making multiple requests to the server to obtain user preferences:

Here you can see multiple error reports being submitted without user consent

### Server configuration

- Operating system: Docker
- Web server: apache 2
- Database version: mariadb 10.3
- PHP version: 7.4
- phpMyAdmin version: a6422ced622b9e39024f09d4148d65f429b9aa4a

### Client configuration

- Browser: Latest Chrome
- Operating system: OSX 10.14

### PR DESCRIPTION
**Pull request by shucon**
Signed-off-by: Saksham Gupta <shucon01@gmail.com>

### Description

Fixes: #16568

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

**Review by williamdes — APPROVED** [review]
Looks nice !
Should maybe sent to QA_5_1 ?

File: js/src/error_report.js [hunk 1]
```diff
@@ -10,11 +10,42 @@ var ErrorReport = {
...
      * @return void
      */
+    errorDataHandler: function (data, exception) {
+        if (data.success !== true) {
+            Functions.ajaxShowMessage(data.error, false);
+            return;
+        }
+        if (data.report_setting === 'ask') {
+            ErrorReport.showErrorNotification();
+        } else if (data.report_setting === 'always') {
```
Comment (williamdes) [inline js/src/error_report.js:33]: could be if and return and another if, but you can let it like that

Comment (shucon) [reply js/src/error_report.js:33]: It's like the way it was before, just in another function.

**Comment by shucon** [comment]
> Looks nice !
> Should maybe sent to QA_5_1 ?

I just looked at QA_5_1 and I think it would be better to keep it in master. There are some major changes in the `error_report.js` file and the issue is also reproducible in only the master branch.

**Pull request merged**